app/oracle.py

```diff
@@ -165,53 +181,82 @@ def prompt(prompt_message, prompt_end):
             continue
 
 
+def get_previous_frame():
+    """Since the contract lacks a method that returns the time of last report and the reported numbers
+    we are using web3.py filtering to fetch it from the contract events."""
+    logging.info('Getting previously reported numbers (will be fetched from events)...')
+    SECONDS_PER_ETH1_BLOCK = 14
+    latest_block = w3.eth.getBlock('latest')
+    #Calculate earliest block to limit scanning depth
+    from_block = int((latest_block['timestamp']-genesis_time)/SECONDS_PER_ETH1_BLOCK)
+    events = oracle.events.Completed.getLogs(fromBlock=from_block, toBlock='latest')
+    if events:
+        event = events[-1]
+        epoch = event['args']['epochId']
+        balance = event['args']['beaconBalance']
+        beacon_validators = event['args']['beaconValidators']
+        timestamp = w3.eth.getBlock(event['blockHash'])['timestamp']
+        return (epoch, balance, beacon_validators, timestamp)
+    else:
+        logging.warning('No events on the contract')
+        return False
+
```

Comment:
according to PEP 8: 2 blank lines are expected after class or function definition